src/scene/materials/standard-material-options-builder.js

```diff
@@ -0,0 +1,317 @@
+Object.assign(pc, function () {
+    var StandardMaterialOptionsBuilder = function () {
+    };
+
+    StandardMaterialOptionsBuilder.prototype.update = function (device, scene, stdMat, objDefs, staticLightList, pass, sortedLights, prefilteredCubeMap128) {
+
+        var options = {
+            pass: pass,
+            alphaTest: stdMat.alphaTest > 0,
+            forceFragmentPrecision: stdMat.forceFragmentPrecision,
+            chunks: stdMat.chunks,
+            blendType: stdMat.blendType,
+            forceUv1: stdMat.forceUv1
+        };
+
+        if (objDefs) {
+            options.screenSpace = (objDefs & pc.SHADERDEF_SCREENSPACE) !== 0;
+            options.skin = (objDefs & pc.SHADERDEF_SKIN) !== 0;
+            options.useInstancing = (objDefs & pc.SHADERDEF_INSTANCING) !== 0;
+        }
+
+        var minimalOptions = pass > pc.SHADER_FORWARDHDR && pass <= pc.SHADER_PICK;
+
+        if (minimalOptions) {
+            var optionsMin = this._generateMinOptions(stdMat);
+            Object.assign(options, optionsMin);
+        } else {
+            options.useTexCubeLod = device.useTexCubeLod;
+
+            var envOptions = this._generateEnvOptions(stdMat, scene, prefilteredCubeMap128);
+            Object.assign(options, envOptions);
+
+            var optionsMat = this._generateMaterialOptions(stdMat);
+            Object.assign(options, optionsMat);
+
+            if (pass === pc.SHADER_FORWARDHDR) {
+                if (options.gamma) options.gamma = pc.GAMMA_SRGBHDR;
+                options.toneMap = pc.TONEMAP_LINEAR;
+            }
+
+            if (objDefs) {
+                if (stdMat.normalMap)
+                    options.hasTangents = (objDefs & pc.SHADERDEF_TANGENTS) !== 0;
+            }
+
+            var optionsLight = this._generateLightOptions(stdMat, objDefs, sortedLights, staticLightList);
+            Object.assign(options, optionsLight);
+        }
+
+        var hasUv0 = false;
+        var hasUv1 = false;
+        var hasVcolor = false;
+        if (objDefs) {
+            hasUv0 = (objDefs & pc.SHADERDEF_UV0) !== 0;
+            hasUv1 = (objDefs & pc.SHADERDEF_UV1) !== 0;
+            hasVcolor = (objDefs & pc.SHADERDEF_VCOLOR) !== 0;
+        }
+
+        this._mapXForms = [];
+        for (var p in pc._matTex2D) {
```

Comment:
Probably doesn't need addressing in this PR, but I hate how the map names have been stored directly in the pc namespace. Would be great to remove that pollution at some point.

Reply:
This structure is being used only in standard.js and standard-material.js, so it's easy to remove it from pc namespace, just don't know where exactly it should be.